src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -841,6 +863,30 @@ public void testCacheOnViolationSuppression() throws Exception {
         }
     }
 
+    @Test
+    public void testHaltOnException() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(CheckWhichThrowsError.class);
+
+        final DefaultConfiguration treeWalkerConfig = createModuleConfig(TreeWalker.class);
+        treeWalkerConfig.addChild(checkConfig);
+
+        final DefaultConfiguration checkerConfig = createRootConfig(treeWalkerConfig);
+        checkerConfig.addChild(treeWalkerConfig);
```

Comment:
This line isn't necessary, `createRootConfig` already adds the child you give.
See https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java#L176 .